shared/src/main/scala/MiscHelpers.scala

```diff
@@ -28,6 +28,26 @@ object MiscHelpers:
     case (a: VNum, b: String)   => a.toString.compareTo(b)
     case (a: String, b: String) => a.compareTo(b)
 
+  def compareExact(
+      a: VAny,
+      b: VAny
+  )(using ctx: Context): Int = (a, b) match
+    case (a: VVal, b: VVal) => compare(a, b)
```

Comment:
I'm thinking we might want to get rid of `VVal` and `VAtom`. It's not immediately clear to someone new to our codebase what that means (in fact, I have to take a few seconds to remember what they are every time I see them, and I wrote them)